src/pool-cl/CLPositionManager.sol

```diff
@@ -343,18 +337,28 @@ contract CLPositionManager is
         address owner,
         bytes calldata hookData
     ) internal {
-        (uint160 sqrtPriceX96,,,) = clPoolManager.getSlot0(poolKey.toId());
-
         // Use the credit on the pool manager as the amounts for the mint.
-        uint256 liquidity = LiquidityAmounts.getLiquidityForAmounts(
+        uint256 liquidity =
+            _getFullCreditLiquidity(poolKey.toId(), poolKey.currency0, poolKey.currency1, tickLower, tickUpper);
+
+        _mint(poolKey, tickLower, tickUpper, liquidity, amount0Max, amount1Max, owner, hookData);
+    }
+
+    function _getFullCreditLiquidity(
+        PoolId id,
+        Currency currency0,
```

Comment:
Because poolKey in _mintFromDeltas is calldata , and poolKey in _increase is memory , so can not directly pass poolKey